src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseServiceImpl.java

```diff
@@ -323,66 +300,36 @@ private void transitionCaseGroupStatus(final Case targetCase, final CaseEvent ca
     }
   }
 
-  /**
-   * This has been triggered by a 'RESPONDENT_ENROLED' event. If we find any associated Case Groups
-   * with only B cases we need to make case 'INACTIONABLE' and create BI case.
-   *
-   * @param category a category - currently only called for RESPONDENT_ENROLED
-   * @param caseGroups a list of case groups
-   * @param caseEvent a case event
-   * @param newCase a case to provide a party id
-   * @throws CTPException thrown if database error etc
-   */
-  private void processCaseCreationAndTransitionsDuringEnrolment(
-      final Category category,
-      final List<CaseGroup> caseGroups,
-      final CaseEvent caseEvent,
-      final Case newCase)
+  private void processActionPlanChange(final Case targetCase, final boolean enrolments)
       throws CTPException {
 
+    List<CaseGroup> caseGroups =
+        caseGroupService.findCaseGroupsForExecutedCollectionExercises(targetCase);
+
     for (CaseGroup caseGroup : caseGroups) {
 
       // fetch all B and BI cases associated to the case group being processed
       List<Case> cases =
           caseRepo.findByCaseGroupFKOrderByCreatedDateTimeDesc(caseGroup.getCaseGroupPK());
 
-      // Create a new BI case for the respondent enrolling (if one doesn't already exit)
-      // This is primarily to guard against multiple RESPONDENT_ENROLED case events for the same
-      // user
-      // caused by a double click scenario in the verification email journey
-      List<Case> biCases =
-          cases
-              .stream()
-              .filter(
-                  c ->
-                      c.getSampleUnitType().toString().equals("BI")
-                          && c.getPartyId().equals(newCase.getPartyId()))
-              .collect(Collectors.toList());
-      if (biCases.size() != 0) {
-        log.warn(
-            "Existing BI case found during enrolment for "
-                + "partyid: {} for casegroup: {} with caseid: {}",
-            newCase.getPartyId().toString(),
-            caseGroup.getId(),
-            biCases.get(0).getId());
-      } else {
-        Case c = new Case();
-        c.setPartyId(newCase.getPartyId());
-        createNewCase(category, caseEvent, cases.get(0), c);
-        log.info(
-            "BI case created during enrolment for partyid: {} for casegroup: {}",
-            newCase.getPartyId().toString(),
-            caseGroup.getId());
+      List<ActionPlanDTO> actionPlans =
+          actionSvcClientService.getActionPlans(caseGroup.getCollectionExerciseId(), enrolments);
+
+      if (actionPlans == null || actionPlans.size() != 1) {
+        log.error(
+            "One action plan expected for collectionExerciseId={} with active enrolment",
```

Comment:
Log line shouldn't reference `active` enrolment
This function can be called for both active and inactive enrolments